specs/sharding/beacon-chain.md

```diff
@@ -103,786 +87,558 @@ The following values are (non-configurable) constants used throughout the specif
-        delta = max(1, prev_price * (samples_length - TARGET_SAMPLES_PER_BLOB) // TARGET_SAMPLES_PER_BLOB // adjustment_quotient)
-        return min(prev_price + delta, MAX_SAMPLE_PRICE)
-    else:
-        delta = max(1, prev_price * (TARGET_SAMPLES_PER_BLOB - samples_length) // TARGET_SAMPLES_PER_BLOB // adjustment_quotient)
-        return max(prev_price, MIN_SAMPLE_PRICE + delta) - delta
+def verify_degree_proof(commitment: KZGCommitment, degree: uint64, proof: KZGCommitment):
+    """
+    Verifies that the commitment is of polynomial degree <= degree. 
+    """
+
+    assert (
+        bls.Pairing(proof, G2_SETUP[0])
+        == bls.Pairing(commitment, G2_SETUP[-degree - 1])
+    )
 ```
 
-#### `compute_committee_source_epoch`
+#### `block_to_field_elements`
 
 ```python
-def compute_committee_source_epoch(epoch: Epoch, period: uint64) -> Epoch:
+def block_to_field_elements(block: bytes) -> List[BLSFieldElement]:
     """
-    Return the source epoch for computing the committee.
+    Slices a block into 31 byte chunks that can fit into field elements
     """
-    source_epoch = Epoch(epoch - epoch % period)
-    if source_epoch >= period:
-        source_epoch -= period  # `period` epochs lookahead
-    return source_epoch
+    sliced_block = [block[i:i + 31] for i in range(0, len(bytes), 31)]
+    return [BLSFieldElement(int.from_bytes(x, "little")) for x in sliced_block]
 ```
 
-#### `batch_apply_participation_flag`
+#### `roots_of_unity`
 
 ```python
-def batch_apply_participation_flag(state: BeaconState, bits: Bitlist[MAX_VALIDATORS_PER_COMMITTEE],
-                                   epoch: Epoch, full_committee: Sequence[ValidatorIndex], flag_index: int):
-    if epoch == get_current_epoch(state):
-        epoch_participation = state.current_epoch_participation
-    else:
-        epoch_participation = state.previous_epoch_participation
-    for bit, index in zip(bits, full_committee):
-        if bit:
-            epoch_participation[index] = add_flag(epoch_participation[index], flag_index)
-```
+def roots_of_unity(order: uint64) -> List[BLSFieldElement]:
+    r = []
+    root_of_unity = pow(PRIMITIVE_ROOT_OF_UNITY, (MODULUS - 1) // order, MODULUS)
 
-### Beacon state accessors
+    current_root_of_unity = 1
+    for i in range(len(SAMPLES_PER_BLOB * FIELD_ELEMENTS_PER_SAMPLE)):
+        r.append(current_root_of_unity)
+        current_root_of_unity = current_root_of_unity * root_of_unity % MODULUS
+    return r
+```
 
-#### Updated `get_committee_count_per_slot`
+#### `modular_inverse`
 
 ```python
-def get_committee_count_per_slot(state: BeaconState, epoch: Epoch) -> uint64:
-    """
-    Return the number of committees in each slot for the given ``epoch``.
-    """
-    return max(uint64(1), min(
-        get_active_shard_count(state, epoch),
-        uint64(len(get_active_validator_indices(state, epoch))) // SLOTS_PER_EPOCH // TARGET_COMMITTEE_SIZE,
-    ))
+def modular_inverse(a):
+    assert(a == 0):
```

Comment:
Typo (assert a != 0).

Otherwise, the hash-to-curve spec enforces 0⁻¹ = 0 (mod p) so it's not a cost to do this here. And traditional modular inversion techniques (Little Fermat Theorem or binary Extended Euclidean Algorithm) will return 0 on input 0.